src/utils/transition-manager.js

```diff
@@ -132,22 +138,30 @@ export default class TransitionManager {
       isRotating: startProps.bearing !== endProps.bearing ||
         startProps.pitch !== endProps.pitch
     };
-
+    const currentTime = Date.now();
+    let newDuration = endProps.transitionDuration;
+    let newEasing = endProps.transitionEasing;
+    let newInterpolator = endProps.transitionInterpolator;
+    if (this.state.interruption === TRANSITION_EVENTS.UPDATE) {
```

Comment:
This should be moved into `processViewportChange`, where `TRANSITION_EVENTS .SNAP_TO_END` is handled